packages/ai/tests/ai/web/test_server.py

```
@@ -0,0 +1,140 @@
+_mock_endpoints = MagicMock()
+for _name in ('use', 'ping', 'version', 'shutdown', 'status'):
+    setattr(_mock_endpoints, _name, MagicMock())
+sys.modules.setdefault('ai.web.endpoints', _mock_endpoints)
+
+# ai.web.denied (server.py imports from .denied)
+sys.modules.setdefault('ai.web.denied', MagicMock())
+
+# ai.constants
+_mock_constants = MagicMock()
+_mock_constants.CONST_DEFAULT_WEB_PORT = 5565
+_mock_constants.CONST_DEFAULT_WEB_HOST = '0.0.0.0'
```

Ruff (0.15.7)

[error] Possible binding to all interfaces

(S104)